doc/whats_new/v1.1.rst

```diff
@@ -181,6 +181,24 @@ Changelog
   get accurate results when the number of features is large.
   :pr:`21109` by :user:`Smile <x-shadow-man>`.
 
+- |Enhancement| The :class:`decomposition.MiniBatchDictionaryLearning` and
+  :func:`decomposition.dict_learning_online` have been refactored and now have a
+  stopping criterion based on a small change of the dictionary or objective function,
```

Comment:
I did not see an additional test where we test specifically the stopping criterion.
It could be worth adding one maybe by calling only the function with two dictionaries that are close to each other already.

Reply:
There's test_dict_learning_online_verbosity. It's designed to trigger the 2 early stopping strats. We test them indirectly by checking that they are covered.

Reply:
Good for me